openclaw-advanced-agent-security-guide.html

```
@@ -0,0 +1,2553 @@
+        <h3 class="sub-title">3.3 構造化監査ログの書き込み</h3>
+        <div class="code-block">
+          <div class="code-label"><i class="ti ti-brand-typescript"></i> plugin.ts — agent_end</div>
+          <pre><code class="language-typescript">export const agentEnd: AgentEndHook = async (ctx) => {
+  const { sessionKey, runId, toolCalls, duration } = ctx;
+
+  await appendFile("/var/log/openclaw/audit.jsonl", JSON.stringify({
+    timestamp: new Date().toISOString(),
+    sessionKey,
+    runId,
+    toolCallCount: toolCalls.length,
+    toolNames: toolCalls.map(t => t.name),
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ > ].

(spec-char-escape)